app/controllers/users/fulfillment_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Users::FulfillmentController < ApplicationController
+  before_action :require_login
+
+  def index
+    my_order_items= OrderItem.where(product_id: current_user.products)
+    
+    @grouped_by_status = my_order_items.group_by { |item| item.order.order_state}
+
+    my_orders = my_order_items.pluck(:order_id)
+
+    my_orders = Order.where(id: my_orders)
```

Comment:
For clarity you may want the variables in line 9 & line 11 to be different.
I recommend like 9 be something like `my_order_ids`